File: BONUS/menu.py
#!/usr/bin/env python3
"""
Interactive Terminal Menu for Small-World Network Analysis
"""

import os
import sys
import subprocess
import time

# ANSI color codes
class Colors:
    HEADER = '\033[95m'
    BLUE = '\033[94m'
    CYAN = '\033[96m'
    GREEN = '\033[92m'
    YELLOW = '\033[93m'
    RED = '\033[91m'
    BOLD = '\033[1m'
    UNDERLINE = '\033[4m'
    END = '\033[0m'

def clear_screen():
    os.system('clear' if os.name == 'posix' else 'cls')

def print_header():
    clear_screen()
    print(f"{Colors.CYAN}{Colors.BOLD}")
    print("╔══════════════════════════════════════════════════════════════╗")
    print("║       SMALL-WORLD NETWORK PHENOMENA ANALYZER                 ║")
    print("║       Real-World Simulations & Visualizations                ║")
    print("╚══════════════════════════════════════════════════════════════╝")
    print(f"{Colors.END}\n")

def print_menu():
    print(f"{Colors.BOLD}📊 MAIN MENU{Colors.END}\n")
    print(f"{Colors.GREEN}[1]{Colors.END} 🌐 Generate Network Datasets (Python)")
    print(f"{Colors.GREEN}[2]{Colors.END} 🔬 Analyze Network Metrics (C, L)")
    print(f"{Colors.GREEN}[3]{Colors.END} 🦠 Disease Spread Simulation")
    print(f"{Colors.GREEN}[4]{Colors.END} 📱 Social Influence & Viral Marketing")
    print(f"{Colors.GREEN}[5]{Colors.END} 🤝 Cooperation Evolution (Game Theory)")
    print(f"{Colors.GREEN}[6]{Colors.END} 🚗 Transport & Navigation Efficiency")
    print(f"{Colors.GREEN}[7]{Colors.END} 📊 Compare All Network Types")
    print(f"{Colors.GREEN}[8]{Colors.END} 🌐 Launch Web Dashboard (Recommended!)")
    print(f"{Colors.GREEN}[9]{Colors.END} 📖 View README Documentation")
    print(f"{Colors.RED}[0]{Colors.END} 🚪 Exit\n")

def run_command(cmd, description):
    print(f"\n{Colors.YELLOW}➜ {description}{Colors.END}")
    print(f"{Colors.CYAN}Command: {cmd}{Colors.END}\n")
    result = subprocess.run(cmd, shell=True)
    print(f"\n{Colors.BLUE}{'='*60}{Colors.END}")
    input(f"\n{Colors.BOLD}Press Enter to continue...{Colors.END}")
    return result.returncode == 0

def generate_networks():
    print_header()
    print(f"{Colors.BOLD}🌐 NETWORK GENERATION{Colors.END}\n")
    print("This will generate three network types:")
    print("  • Watts-Strogatz (Small-World)")
    print("  • Erdős-Rényi (Random)")
    print("  • Barabási-Albert (Scale-Free)")
    print("\nNote: Requires Python packages: networkx, numpy, pandas, matplotlib")
    print(f"{Colors.YELLOW}Install with: pip install networkx numpy pandas matplotlib{Colors.END}\n")
    
    proceed = input("Continue? (y/n): ").strip().lower()
    if proceed == 'y':
        run_command("python3 generate.py", "Generating networks...")

def analyze_networks():
    print_header()
    print(f"{Colors.BOLD}🔬 NETWORK ANALYSIS{Colors.END}\n")
    print("Select network to analyze:")
    print("[1] Watts-Strogatz (Small-World)")
    print("[2] Erdős-Rényi (Random)")
    print("[3] Barabási-Albert (Scale-Free)")
    print("[4] All Networks")
    
    choice = input("\nEnter choice (1-4): ").strip()
    
    data_dir = "small_world_analysis_data"
    networks = {
        '1': ('data_proof_WS.txt', 'Watts-Strogatz'),
        '2': ('data_proof_ER.txt', 'Erdős-Rényi'),
        '3': ('data_proof_BA.txt', 'Barabási-Albert')
    }
    
    if choice in networks:
        file, name = networks[choice]
        cmd = f"./network_analyzer {data_dir}/{file}"
        run_command(cmd, f"Analyzing {name} network...")
    elif choice == '4':
        for key in ['1', '2', '3']:
            file, name = networks[key]
            cmd = f"./network_analyzer {data_dir}/{file}"
            run_command(cmd, f"Analyzing {name} network...")

def disease_simulation():
    print_header()
    print(f"{Colors.BOLD}🦠 DISEASE SPREAD SIMULATION{Colors.END}\n")
    print("SIR Model: Susceptible → Infected → Recovered")
    print("\nSelect network:")
    print("[1] Small-World (WS) - Fast spread")
    print("[2] Random (ER) - Moderate spread")
    print("[3] Scale-Free (BA) - Very fast spread")
    
    choice = input("\nEnter choice (1-3): ").strip()
    
    networks = {
        '1': 'data_proof_WS.txt',
        '2': 'data_proof_ER.txt',
        '3': 'data_proof_BA.txt'
    }
    
    if choice in networks:
        print(f"\n{Colors.CYAN}Parameters:{Colors.END}")
        infection = input("Infection probability (default 0.3): ").strip() or "0.3"
        recovery = input("Recovery probability (default 0.1): ").strip() or "0.1"
        sims = input("Number of simulations (default 10): ").strip() or "10"
        
        cmd = f"./disease_spread small_world_analysis_data/{networks[choice]} {infection} {recovery} {sims}"
        run_command(cmd, "Running disease spread simulation...")

def social_influence():
    print_header()
    print(f"{Colors.BOLD}📱 SOCIAL INFLUENCE SIMULATION{Colors.END}\n")
    print("Select simulation mode:")
    print("[1] Threshold Model - Peer pressure adoption")
    print("[2] Viral Marketing - Influencer targeting")
    
    mode_choice = input("\nEnter choice (1-2): ").strip()
    
    mode = 'threshold' if mode_choice == '1' else 'viral'
    
    print("\nSelect network:")
    print("[1] Small-World (WS)")
    print("[2] Random (ER)")
    print("[3] Scale-Free (BA)")
    
    net_choice = input("\nEnter choice (1-3): ").strip()
    
    networks = {
        '1': 'data_proof_WS.txt',
        '2': 'data_proof_ER.txt',
        '3': 'data_proof_BA.txt'
    }
    
    if net_choice in networks:
        cmd = f"./social_influence small_world_analysis_data/{networks[net_choice]} {mode}"
        run_command(cmd, f"Running {mode} simulation...")

def cooperation_simulation():
    print_header()
    print(f"{Colors.BOLD}🤝 COOPERATION EVOLUTION{Colors.END}\n")
    print("Prisoner's Dilemma on Networks")
    print("\nSelect network:")
    print("[1] Small-World (WS) - High cooperation")
    print("[2] Random (ER) - Low cooperation")
    print("[3] Scale-Free (BA) - Variable cooperation")
    
    choice = input("\nEnter choice (1-3): ").strip()
    
    networks = {
        '1': 'data_proof_WS.txt',
        '2': 'data_proof_ER.txt',
        '3': 'data_proof_BA.txt'
    }
    
    if choice in networks:
        print(f"\n{Colors.CYAN}Parameters:{Colors.END}")
        init_coop = input("Initial cooperation rate (default 0.5): ").strip() or "0.5"
        gens = input("Generations (default 100): ").strip() or "100"
        
        cmd = f"./cooperation small_world_analysis_data/{networks[choice]} {init_coop} {gens}"
        run_command(cmd, "Running cooperation evolution...")

def transport_simulation():
    print_header()
    print(f"{Colors.BOLD}🚗 TRANSPORT SIMULATION{Colors.END}\n")
    print("Select analysis mode:")
    print("[1] Routing Efficiency - Packet delivery")
    print("[2] Hub Vulnerability - Critical node analysis")
    
    mode_choice = input("\nEnter choice (1-2): ").strip()
    
    mode = 'routing' if mode_choice == '1' else 'vulnerability'
    
    print("\nSelect network:")
    print("[1] Small-World (WS)")
    print("[2] Random (ER)")
    print("[3] Scale-Free (BA)")
    
    net_choice = input("\nEnter choice (1-3): ").strip()
    
    networks = {
        '1': 'data_proof_WS.txt',
        '2': 'data_proof_ER.txt',
        '3': 'data_proof_BA.txt'
    }
    
    if net_choice in networks:
        cmd = f"./transport small_world_analysis_data/{networks[net_choice]} {mode}"
        run_command(cmd, f"Running {mode} analysis...")

def compare_all():
    print_header()
    print(f"{Colors.BOLD}📊 COMPARING ALL NETWORK TYPES{Colors.END}\n")
    print("This will run disease spread on all three networks for comparison.\n")
    
    proceed = input("Continue? (y/n): ").strip().lower()
    if proceed == 'y':
        data_dir = "small_world_analysis_data"
        networks = [
            ('data_proof_WS.txt', 'Watts-Strogatz (Small-World)'),
            ('data_proof_ER.txt', 'Erdős-Rényi (Random)'),
            ('data_proof_BA.txt', 'Barabási-Albert (Scale-Free)')
        ]
        
        for file, name in networks:
            print(f"\n{Colors.YELLOW}{'='*60}{Colors.END}")
            print(f"{Colors.BOLD}Network: {name}{Colors.END}")
            print(f"{Colors.YELLOW}{'='*60}{Colors.END}\n")
            cmd = f"./disease_spread {data_dir}/{file} 0.3 0.1 5"
            subprocess.run(cmd, shell=True)
            time.sleep(2)
        
        input(f"\n{Colors.BOLD}Press Enter to continue...{Colors.END}")

def launch_dashboard():
    print_header()
    print(f"{Colors.BOLD}🌐 LAUNCHING WEB DASHBOARD{Colors.END}\n")
    print("Starting Flask web server...")
    print(f"\n{Colors.GREEN}The dashboard will open in your browser at: http://localhost:8080{Colors.END}")
    print(f"{Colors.YELLOW}Press Ctrl+C to stop the server{Colors.END}\n")
    
    time.sleep(2)
    
    try:
        subprocess.run("python3 web_dashboard.py", shell=True)
    except KeyboardInterrupt:
        print(f"\n\n{Colors.RED}Server stopped.{Colors.END}")
        input(f"\n{Colors.BOLD}Press Enter to continue...{Colors.END}")

def view_readme():
    print_header()
    run_command("cat README.md | less", "Viewing README documentation...")

def check_prerequisites():
    """Check if necessary files and programs exist"""
    issues = []
    
    # Check for compiled programs
    programs = ['network_analyzer', 'disease_spread', 'social_influence', 'cooperation', 'transport']
    for prog in programs:
        if not os.path.exists(prog):
            issues.append(f"Missing compiled program: {prog}")
    
    # Check for data directory
    if not os.path.exists('small_world_analysis_data'):
        issues.append("Data directory not found. Run option [1] to generate networks first.")
    
    return issues

def main():
    while True:
        print_header()
        
        # Check prerequisites
        issues = check_prerequisites()
        if issues:
            print(f"{Colors.YELLOW}⚠️  Setup Required:{Colors.END}")
            for issue in issues:
                print(f"   • {issue}")
            print(f"\n{Colors.CYAN}To compile programs, run: ./demo.sh{Colors.END}")
            print(f"{Colors.CYAN}To generate data, select option [1]{Colors.END}\n")
        
        print_menu()
        
        choice = input(f"{Colors.BOLD}Enter your choice (0-9): {Colors.END}").strip()
        
        if choice == '0':
            print_header()
            print(f"{Colors.GREEN}Thank you for using Small-World Network Analyzer!{Colors.END}\n")
            sys.exit(0)
        elif choice == '1':
            generate_networks()
        elif choice == '2':
            analyze_networks()
        elif choice == '3':
            disease_simulation()
        elif choice == '4':
            social_influence()
        elif choice == '5':
            cooperation_simulation()
        elif choice == '6':
            transport_simulation()
        elif choice == '7':
            compare_all()
        elif choice == '8':
            launch_dashboard()
        elif choice == '9':
            view_readme()
        else:
            print(f"\n{Colors.RED}Invalid choice. Please try again.{Colors.END}")
            time.sleep(1)

if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print(f"\n\n{Colors.RED}Program interrupted by user.{Colors.END}")
        sys.exit(0)

File: BONUS/menu.cpp
// Interactive Menu System for Small-World Network Analysis
// Provides a user-friendly interface for running all simulations

#include <iostream>
#include <string>
#include <vector>
#include <cstdlib>
#include <sstream>
#include <fstream>
#include <sys/stat.h>

using namespace std;

// ANSI color codes
#define RESET   "\033[0m"
#define BOLD    "\033[1m"
#define RED     "\033[31m"
#define GREEN   "\033[32m"
#define YELLOW  "\033[33m"
#define BLUE    "\033[34m"
#define MAGENTA "\033[35m"
#define CYAN    "\033[36m"

bool fileExists(const string& filename) {
    struct stat buffer;
    return (stat(filename.c_str(), &buffer) == 0);
}

bool directoryExists(const string& dirname) {
    struct stat buffer;
    return (stat(dirname.c_str(), &buffer) == 0 && S_ISDIR(buffer.st_mode));
}

void clearScreen() {
    #ifdef _WIN32
        system("cls");
    #else
        system("clear");
    #endif
}

void printHeader() {
    cout << CYAN << BOLD;
    cout << "\n╔════════════════════════════════════════════════════════════════╗\n";
    cout << "║                                                                ║\n";
    cout << "║     SMALL-WORLD NETWORK PHENOMENA: REAL-WORLD SIMULATIONS     ║\n";
    cout << "║                                                                ║\n";
    cout << "║        Explore disease spread, social influence, and more!    ║\n";
    cout << "║                                                                ║\n";
    cout << "╚════════════════════════════════════════════════════════════════╝\n";
    cout << RESET << "\n";
}

void printMenu() {
    cout << BOLD << "\n━━━━━━━━━━━━━━━━━━━━━ MAIN MENU ━━━━━━━━━━━━━━━━━━━━━\n" << RESET;
    cout << "\n" << YELLOW << "📊 DATA GENERATION" << RESET << "\n";
    cout << "  1. Generate Network Datasets (Python)\n";
    cout << "  2. Check Data Status\n";
    
    cout << "\n" << GREEN << "📈 NETWORK ANALYSIS" << RESET << "\n";
    cout << "  3. Analyze Network Metrics (C & L)\n";
    cout << "  4. Compare All Network Types\n";
    
    cout << "\n" << RED << "🦠 REAL-WORLD SIMULATIONS" << RESET << "\n";
    cout << "  5. Disease Spread (Epidemic Model)\n";
    cout << "  6. Social Influence & Viral Marketing\n";
    cout << "  7. Cooperation Evolution (Game Theory)\n";
    cout << "  8. Transport & Routing Efficiency\n";
    cout << "  9. Hub Vulnerability Analysis\n";
    
    cout << "\n" << MAGENTA << "🌐 WEB DASHBOARD" << RESET << "\n";
    cout << "  10. Launch Web Dashboard (Interactive)\n";
    cout << "  11. Generate HTML Report\n";
    
    cout << "\n" << BLUE << "📚 HELP & INFO" << RESET << "\n";
    cout << "  12. About Small-World Networks\n";
    cout << "  13. View README\n";
    
    cout << "\n" << "  0. Exit\n";
    cout << "\n" << BOLD << "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\n" << RESET;
}

string selectNetworkType() {
    cout << "\n" << YELLOW << "Select Network Type:" << RESET << "\n";
    cout << "  1. Watts-Strogatz (Small-World)\n";
    cout << "  2. Erdős-Rényi (Random)\n";
    cout << "  3. Barabási-Albert (Scale-Free)\n";
    cout << "\nChoice [1-3]: ";
    
    int choice;
    cin >> choice;
    
    switch(choice) {
        case 1: return "small_world_analysis_data/data_proof_WS.txt";
        case 2: return "small_world_analysis_data/data_proof_ER.txt";
        case 3: return "small_world_analysis_data/data_proof_BA.txt";
        default: return "small_world_analysis_data/data_proof_WS.txt";
    }
}

void generateData() {
    cout << "\n" << CYAN << "═══ Generating Network Datasets ═══" << RESET << "\n\n";
    cout << "Running Python script to generate networks...\n\n";
    
    int result = system("python3 generate.py");
    
    if (result == 0) {
        cout << "\n" << GREEN << "✓ Data generated successfully!" << RESET << "\n";
    } else {
        cout << "\n" << RED << "✗ Error: Please install required packages:" << RESET << "\n";
        cout << "  pip install networkx numpy pandas matplotlib\n";
    }
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void checkDataStatus() {
    cout << "\n" << CYAN << "═══ Data Status Check ═══" << RESET << "\n\n";
    
    bool dirExists = directoryExists("small_world_analysis_data");
    
    if (!dirExists) {
        cout << RED << "✗ Data directory not found!" << RESET << "\n";
        cout << "  Please run option 1 to generate data.\n";
    } else {
        cout << GREEN << "✓ Data directory exists" << RESET << "\n\n";
        
        vector<string> files = {"data_proof_WS.txt", "data_proof_ER.txt", "data_proof_BA.txt"};
        vector<string> names = {"Watts-Strogatz (WS)", "Erdős-Rényi (ER)", "Barabási-Albert (BA)"};
        
        for (size_t i = 0; i < files.size(); i++) {
            string path = "small_world_analysis_data/" + files[i];
            if (fileExists(path)) {
                cout << "  " << GREEN << "✓" << RESET << " " << names[i] << ": " << path << "\n";
            } else {
                cout << "  " << RED << "✗" << RESET << " " << names[i] << ": NOT FOUND\n";
            }
        }
    }
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void analyzeMetrics() {
    string file = selectNetworkType();
    
    if (!fileExists(file)) {
        cout << RED << "\n✗ Error: Data file not found!" << RESET << "\n";
        cout << "Please generate data first (option 1).\n";
        cout << "\nPress Enter to continue...";
        cin.ignore();
        cin.get();
        return;
    }
    
    cout << "\n" << CYAN << "═══ Running Network Analysis ═══" << RESET << "\n\n";
    string cmd = "./network_analyzer " + file;
    system(cmd.c_str());
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void compareNetworks() {
    cout << "\n" << CYAN << "═══ Comparing All Network Types ═══" << RESET << "\n\n";
    
    vector<string> files = {"data_proof_WS.txt", "data_proof_ER.txt", "data_proof_BA.txt"};
    vector<string> names = {"WATTS-STROGATZ (Small-World)", "ERDŐS-RÉNYI (Random)", "BARABÁSI-ALBERT (Scale-Free)"};
    
    for (size_t i = 0; i < files.size(); i++) {
        string path = "small_world_analysis_data/" + files[i];
        if (fileExists(path)) {
            cout << YELLOW << "\n▶ " << names[i] << RESET << "\n";
            string cmd = "./network_analyzer " + path;
            system(cmd.c_str());
        }
    }
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void diseaseSpread() {
    string file = selectNetworkType();
    
    if (!fileExists(file)) {
        cout << RED << "\n✗ Error: Data file not found!" << RESET << "\n";
        cout << "\nPress Enter to continue...";
        cin.ignore();
        cin.get();
        return;
    }
    
    cout << "\n" << CYAN << "═══ Disease Spread Simulation ═══" << RESET << "\n\n";
    cout << "Parameters:\n";
    cout << "  Infection probability (0.0-1.0) [default 0.3]: ";
    
    string input;
    cin.ignore();
    getline(cin, input);
    double infect = input.empty() ? 0.3 : stod(input);
    
    cout << "  Recovery probability (0.0-1.0) [default 0.1]: ";
    getline(cin, input);
    double recover = input.empty() ? 0.1 : stod(input);
    
    cout << "  Number of simulations [default 10]: ";
    getline(cin, input);
    int sims = input.empty() ? 10 : stoi(input);
    
    cout << "\n";
    stringstream cmd;
    cmd << "./disease_spread " << file << " " << infect << " " << recover << " " << sims;
    system(cmd.str().c_str());
    
    cout << "\nPress Enter to continue...";
    cin.get();
}

void socialInfluence() {
    string file = selectNetworkType();
    
    if (!fileExists(file)) {
        cout << RED << "\n✗ Error: Data file not found!" << RESET << "\n";
        cout << "\nPress Enter to continue...";
        cin.ignore();
        cin.get();
        return;
    }
    
    cout << "\n" << CYAN << "═══ Social Influence Simulation ═══" << RESET << "\n\n";
    cout << "Select mode:\n";
    cout << "  1. Threshold Model (peer pressure)\n";
    cout << "  2. Viral Marketing (influencer targeting)\n";
    cout << "Choice [1-2]: ";
    
    int choice;
    cin >> choice;
    
    string mode = (choice == 2) ? "viral" : "threshold";
    
    cout << "\n";
    string cmd = "./social_influence " + file + " " + mode;
    system(cmd.c_str());
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void cooperation() {
    string file = selectNetworkType();
    
    if (!fileExists(file)) {
        cout << RED << "\n✗ Error: Data file not found!" << RESET << "\n";
        cout << "\nPress Enter to continue...";
        cin.ignore();
        cin.get();
        return;
    }
    
    cout << "\n" << CYAN << "═══ Cooperation Evolution ═══" << RESET << "\n\n";
    cout << "Initial cooperation probability (0.0-1.0) [default 0.5]: ";
    
    string input;
    cin.ignore();
    getline(cin, input);
    double prob = input.empty() ? 0.5 : stod(input);
    
    cout << "Number of generations [default 100]: ";
    getline(cin, input);
    int gens = input.empty() ? 100 : stoi(input);
    
    cout << "\n";
    stringstream cmd;
    cmd << "./cooperation " << file << " " << prob << " " << gens;
    system(cmd.str().c_str());
    
    cout << "\nPress Enter to continue...";
    cin.get();
}

void transport() {
    string file = selectNetworkType();
    
    if (!fileExists(file)) {
        cout << RED << "\n✗ Error: Data file not found!" << RESET << "\n";
        cout << "\nPress Enter to continue...";
        cin.ignore();
        cin.get();
        return;
    }
    
    cout << "\n" << CYAN << "═══ Transport Efficiency ═══" << RESET << "\n\n";
    string cmd = "./transport " + file + " routing";
    system(cmd.c_str());
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void vulnerability() {
    string file = selectNetworkType();
    
    if (!fileExists(file)) {
        cout << RED << "\n✗ Error: Data file not found!" << RESET << "\n";
        cout << "\nPress Enter to continue...";
        cin.ignore();
        cin.get();
        return;
    }
    
    cout << "\n" << CYAN << "═══ Hub Vulnerability Analysis ═══" << RESET << "\n\n";
    string cmd = "./transport " + file + " vulnerability";
    system(cmd.c_str());
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void launchDashboard() {
    cout << "\n" << CYAN << "═══ Launching Web Dashboard ═══" << RESET << "\n\n";
    cout << "Starting web server...\n\n";
    
    if (fileExists("dashboard.html")) {
        cout << GREEN << "✓ Opening dashboard in browser..." << RESET << "\n\n";
        #ifdef __APPLE__
            system("open dashboard.html");
        #elif __linux__
            system("xdg-open dashboard.html");
        #else
            system("start dashboard.html");
        #endif
        
        cout << "Dashboard opened in your default browser.\n";
        cout << "You can also manually open: dashboard.html\n";
    } else {
        cout << YELLOW << "! Dashboard file not found. Generating now..." << RESET << "\n";
        system("python3 create_dashboard.py");
    }
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void generateReport() {
    cout << "\n" << CYAN << "═══ Generating HTML Report ═══" << RESET << "\n\n";
    cout << "Creating comprehensive analysis report...\n\n";
    
    system("python3 create_dashboard.py");
    
    cout << "\n" << GREEN << "✓ Report generated: dashboard.html" << RESET << "\n";
    cout << "Open it in your browser to view interactive visualizations.\n";
    
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

void aboutSmallWorld() {
    clearScreen();
    cout << CYAN << BOLD;
    cout << "\n╔════════════════════════════════════════════════════════════════╗\n";
    cout << "║              ABOUT SMALL-WORLD NETWORKS                        ║\n";
    cout << "╚════════════════════════════════════════════════════════════════╝\n";
    cout << RESET << "\n";
    
    cout << YELLOW << "What are Small-World Networks?" << RESET << "\n";
    cout << "Small-world networks are characterized by:\n";
    cout << "  • " << GREEN << "High Clustering (C)" << RESET << " - Strong local communities\n";
    cout << "  • " << GREEN << "Short Path Lengths (L)" << RESET << " - Fast global connectivity\n\n";
    
    cout << YELLOW << "Real-World Examples:" << RESET << "\n";
    cout << "  🧠 Neural networks in the brain\n";
    cout << "  👥 Social networks (6 degrees of separation)\n";
    cout << "  🌐 The Internet and World Wide Web\n";
    cout << "  ⚡ Power grids\n";
    cout << "  🦠 Disease transmission networks\n\n";
    
    cout << YELLOW << "Why Do They Matter?" << RESET << "\n";
    cout << "  • Enable " << GREEN << "efficient information spread" << RESET << "\n";
    cout << "  • Balance " << BLUE << "local + global connectivity" << RESET << "\n";
    cout << "  • Explain " << MAGENTA << "rapid epidemic spread" << RESET << "\n";
    cout << "  • Guide " << CYAN << "viral marketing strategies" << RESET << "\n\n";
    
    cout << YELLOW << "Key Discovery:" << RESET << "\n";
    cout << "Watts & Strogatz (1998) showed that just a few random\n";
    cout << "\"shortcuts\" in a regular network create small-world properties!\n\n";
    
    cout << "Press Enter to continue...";
    cin.ignore();
    cin.get();
}

void viewReadme() {
    clearScreen();
    cout << CYAN << "═══ README Content ═══" << RESET << "\n\n";
    system("cat README.md | head -100");
    cout << "\n\n" << YELLOW << "(Scroll up to see full content)" << RESET << "\n";
    cout << "\nPress Enter to continue...";
    cin.ignore();
    cin.get();
}

int main() {
    int choice;
    
    while (true) {
        clearScreen();
        printHeader();
        printMenu();
        
        cout << "\n" << BOLD << "Enter your choice: " << RESET;
        cin >> choice;
        
        switch (choice) {
            case 0:
                cout << "\n" << GREEN << "Thank you for exploring Small-World Networks! 👋\n" << RESET;
                return 0;
                
            case 1:
                generateData();
                break;
                
            case 2:
                checkDataStatus();
                break;
                
            case 3:
                analyzeMetrics();
                break;
                
            case 4:
                compareNetworks();
                break;
                
            case 5:
                diseaseSpread();
                break;
                
            case 6:
                socialInfluence();
                break;
                
            case 7:
                cooperation();
                break;
                
            case 8:
                transport();
                break;
                
            case 9:
                vulnerability();
                break;
                
            case 10:
                launchDashboard();
                break;
                
            case 11:
                generateReport();
                break;
                
            case 12:
                aboutSmallWorld();
                break;
                
            case 13:
                viewReadme();
                break;
                
            default:
                cout << "\n" << RED << "Invalid choice! Please try again." << RESET << "\n";
                cout << "\nPress Enter to continue...";
                cin.ignore();
                cin.get();
        }
    }
    
    return 0;
}
